options/rtdl/generic/linker.cpp

```diff
@@ -851,6 +860,11 @@ Scope::Scope()
 : _objects(getAllocator()) { }
 
 void Scope::appendObject(SharedObject *object) {
+	if (_objects.size()) {
+		auto prevObject = _objects[_objects.size() - 1];
+		object->linkMap.prev = &prevObject->linkMap;
+		prevObject->linkMap.next = &object->linkMap;
+	}
```

Comment:
Hm, maybe this should be moved to `Loader::submitObject`? Semantically, this happens during loading and the scope is not really involved here since the linkMap is global anyway.